nifi-nar-bundles/nifi-neo4j-bundle/nifi-neo4j-processors/src/main/java/org/apache/nifi/processors/neo4j/Neo4JCypherExecutor.java

```diff
@@ -0,0 +1,203 @@
+@CapabilityDescription("This processor executes a Neo4J Query (https://www.neo4j.com/) defined in the 'Neo4j Query' property of the "
+    + "FlowFile and writes the result to the FlowFile body in JSON format.")
+@WritesAttributes({
+    @WritesAttribute(attribute = AbstractNeo4JCypherExecutor.ERROR_MESSAGE, description = "Neo4J error message"),
+    @WritesAttribute(attribute = AbstractNeo4JCypherExecutor.LABELS_ADDED, description = "Number of labels added"),
+    @WritesAttribute(attribute = AbstractNeo4JCypherExecutor.NODES_CREATED, description = "Number of nodes created"),
+    @WritesAttribute(attribute = AbstractNeo4JCypherExecutor.NODES_DELETED, description = "Number of nodes deleted"),
+    @WritesAttribute(attribute = AbstractNeo4JCypherExecutor.PROPERTIES_SET, description = "Number of properties set"),
+    @WritesAttribute(attribute = AbstractNeo4JCypherExecutor.RELATIONS_CREATED, description = "Number of relationships created"),
+    @WritesAttribute(attribute = AbstractNeo4JCypherExecutor.RELATIONS_DELETED, description = "Number of relationships deleted"),
+    @WritesAttribute(attribute = AbstractNeo4JCypherExecutor.ROWS_RETURNED, description = "Number of rows returned"),
+    })
+public class Neo4JCypherExecutor extends AbstractNeo4JCypherExecutor {
+
+    private static final Set<Relationship> relationships;
+    private static final List<PropertyDescriptor> propertyDescriptors;
+
+    static {
+        final Set<Relationship> tempRelationships = new HashSet<>();
+        tempRelationships.add(REL_SUCCESS);
+        tempRelationships.add(REL_FAILURE);
+        relationships = Collections.unmodifiableSet(tempRelationships);
+
+        final List<PropertyDescriptor> tempDescriptors = new ArrayList<>();
+        tempDescriptors.add(CONNECTION_URL);
+        tempDescriptors.add(USERNAME);
+        tempDescriptors.add(PASSWORD);
+        tempDescriptors.add(QUERY);
+        tempDescriptors.add(LOAD_BALANCING_STRATEGY);
+        tempDescriptors.add(CONNECTION_TIMEOUT);
+        tempDescriptors.add(MAX_CONNECTION_POOL_SIZE);
+        tempDescriptors.add(MAX_CONNECTION_ACQUISITION_TIMEOUT);
+        tempDescriptors.add(IDLE_TIME_BEFORE_CONNECTION_TEST);
+        tempDescriptors.add(MAX_CONNECTION_LIFETIME);
+        tempDescriptors.add(ENCRYPTION);
+        tempDescriptors.add(TRUST_STRATEGY);
+        tempDescriptors.add(TRUST_CUSTOM_CA_SIGNED_CERTIFICATES_FILE);
+
+        propertyDescriptors = Collections.unmodifiableList(tempDescriptors);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        super.onScheduled(context);
+    }
+
```

Comment:
Are there any custom validations for other numbers?  For instance, can you set an invalid number of connection pool?   Are there bounds to the inputs?

Reply:
I did not find any restrictions on connection/timeout values and have updated the default values.